connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -319,9 +319,11 @@ private void poll(long timeoutMs) {
         }
     }
 
-    private void readToLogEnd() {
+    private void readToLogEnd(boolean shouldRetry) {
         Set<TopicPartition> assignment = consumer.assignment();
-        Map<TopicPartition, Long> endOffsets = readEndOffsets(assignment);
+        // readEndOffsets makes listOffsets call to adminClient, if shouldRetry is set to True, the adminClinet
+        // will retry on RetriableExceptions
```

Comment:
Might be worth moving this to a JavaDoc for the method.

Reply:
Yes! Good call.